RestaurantReservation.API/Program.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using RestaurantReservation;
+using RestaurantReservation.Db;
+using System.Text;
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+var tokenValidation = new TokenValidationParameters
+{
+    ValidateIssuer = true,
+    ValidateAudience = true,
+    ValidateLifetime = true,
+    ValidateIssuerSigningKey = true,
+    ValidIssuer = builder.Configuration["JWTToken:Issuer"],
+    ValidAudience = builder.Configuration["JWTToken:Audience"],
+    IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["JWTToken:Key"]))
+};
+
+builder.Services.AddSingleton(tokenValidation);
+
+
+
+
+builder.Services.AddSwaggerGen(c =>
+ {
+
+     // Add security definition for Bearer token
+     c.AddSecurityDefinition("RestaurantApiBearerAuth", new()
+     {
+         Type = SecuritySchemeType.Http,
+         Scheme = "Bearer",
+         Description = "Input a valid token to access this API"
+     });
+
+     c.AddSecurityRequirement(new()
+    {
+        {
+            new ()
+            {
+                Reference = new OpenApiReference {
+                    Type = ReferenceType.SecurityScheme,
+                    Id = "RestaurantApiBearerAuth" }
+            },
+            new List<string>()
+        }
+    });
+ });
+
+
+
+builder.Services.AddAuthorization(op =>
+{
+    op.AddPolicy("Customer", policy =>
+    {
+        policy.RequireAuthenticatedUser();
+        policy.RequireClaim("Type", "Customer");
+    });
+
+    op.AddPolicy("Employee", policy =>
+    {
+        policy.RequireAuthenticatedUser();
+        policy.RequireClaim("Type", "Employee");
+        policy.RequireClaim("Position", Position.Normal.ToString());
+    });
+
+    op.AddPolicy("Manager", policy =>
+    {
+        policy.RequireAuthenticatedUser();
+        policy.RequireClaim("Type", "Employee");
+        policy.RequireClaim("Position", Position.Manager.ToString());
+    });
+
+});
+
+
+
+builder.Services.AddAuthentication("Bearer").AddJwtBearer(p =>
+{
+    p.SaveToken = true;
+    p.TokenValidationParameters = tokenValidation;
+});
+
+
+
+
+
+builder.Services.AddControllers().AddNewtonsoftJson(options =>
+{
+    options.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
+});
+
+builder.Services.AddScoped<RestaurantReservationDbContext>();
```

Comment:
Check the method `AddDbContext`.
And what would the difference be between using it or defining DbContext this way?